onnxruntime/core/session/provider_bridge_ort.cc

```diff
@@ -0,0 +1,563 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// This is the Onnxruntime side of the bridge to allow providers to be built as a DLL
+// It implements onnxruntime::ProviderHost
+
+#include "core/framework/data_types.h"
+#include "core/framework/allocatormgr.h"
+#include "core/providers/dnnl/dnnl_provider_factory.h"
+#include "core/session/abi_session_options_impl.h"
+#include "core/session/ort_apis.h"
+#include "core/platform/env.h"
+#include "core/framework/execution_provider.h"
+#include "core/framework/compute_capability.h"
+#define PROVIDER_BRIDGE_ORT
+#include "core/providers/shared_library/provider_interfaces.h"
+#include "onnx/common/stl_backports.h"
+#include "core/common/logging/logging.h"
+#include "core/common/cpuid_info.h"
+
+namespace onnxruntime {
+
+struct Provider_OrtDevice_Impl : Provider_OrtDevice {
+  OrtDevice v_;
+};
+
+struct Provider_OrtMemoryInfo_Impl : Provider_OrtMemoryInfo {
+  Provider_OrtMemoryInfo_Impl(const char* name_, OrtAllocatorType type_, OrtDevice device_, int id_, OrtMemType mem_type_) : info_{onnxruntime::make_unique<OrtMemoryInfo>(name_, type_, device_, id_, mem_type_)} {}
+
+  std::unique_ptr<OrtMemoryInfo> info_;
+};
+
+struct Provider_IAllocator_Impl : Provider_IAllocator {
+  Provider_IAllocator_Impl(AllocatorPtr p) : p_{p} {}
+
+  void* Alloc(size_t size) override { return p_->Alloc(size); }
+  void Free(void* p) override { return p_->Free(p); }
+
+  AllocatorPtr p_;
+};
+
+struct Provider_IDeviceAllocator_Impl : Provider_IDeviceAllocator {
+  Provider_IDeviceAllocator_Impl(std::unique_ptr<IDeviceAllocator> p) : p_{std::move(p)} {}
+
+  void* Alloc(size_t size) override { return p_->Alloc(size); }
+  void Free(void* p) override { return p_->Free(p); }
+
+  bool AllowsArena() const override { return p_->AllowsArena(); }
+
+  std::unique_ptr<IDeviceAllocator> p_;
+};
+
+struct Provider_TensorProto_Impl : ONNX_NAMESPACE::Provider_TensorProto {
+  Provider_TensorProto_Impl(ONNX_NAMESPACE::TensorProto* p) : p_{p} {}
+
+  void CopyFrom(const Provider_TensorProto& v) override {
+    *p_ = *static_cast<const Provider_TensorProto_Impl*>(&v)->p_;
+  }
+
+  ONNX_NAMESPACE::TensorProto* p_{};
+};  // namespace onnxruntime
```

Comment:
rogue placement of // namespace onnxruntime

Reply:
Ah, there was a couple of those in there. The formatter loves to add those whenever it can.